liquidity/liquidity.go

```diff
@@ -485,6 +571,92 @@ func (m *Manager) makeLoopOutRequest(suggestion *LoopOutRecommendation,
 	}
 }
 
+// worstCaseOutFees calculates the largest possible fees for a loop out swap,
+// comparing the fees for a successful swap to the cost when the client pays
+// the prepay because they failed to sweep the on chain htlc. This is unlikely,
+// because we expect clients to be online to sweep, but we want to account for
+// every outcome so we include it.
+func worstCaseOutFees(prepayRouting, swapRouting, swapFee, minerFee,
+	prepayAmount btcutil.Amount) btcutil.Amount {
+
+	var (
+		successFees = prepayRouting + minerFee + swapFee + swapRouting
+		noShowFees  = prepayRouting + prepayAmount
+	)
+
+	if noShowFees > successFees {
+		return noShowFees
+	}
+
+	return successFees
+}
+
+// existingAutoLoopSummary provides a summary of the existing autoloops which
+// were dispatched during our current budget period.
+type existingAutoLoopSummary struct {
+	// spentFees is the amount we have spent on completed swaps.
+	spentFees btcutil.Amount
+
+	// pendingFees is the worst-case amount of fees we could spend on in
+	// flight autoloops.
+	pendingFees btcutil.Amount
+}
+
+// totalFees returns the total amount of fees that automatically dispatched
+// swaps may consume.
+func (e *existingAutoLoopSummary) totalFees() btcutil.Amount {
+	return e.spentFees + e.pendingFees
+}
+
+// checkExistingAutoLoops calculates the total amount that has been spent by
+// automatically dispatched swaps that have completed, and the worst-case fee
+// total for our set of ongoing, automatically dispatched swaps.
+func (m *Manager) checkExistingAutoLoops(ctx context.Context,
+	loopOuts []*loopdb.LoopOut) (*existingAutoLoopSummary, error) {
+
+	var summary existingAutoLoopSummary
+
+	for _, out := range loopOuts {
+		if out.Contract.Label != labels.AutoOutLabel() {
+			continue
+		}
+
+		// Skip over any swaps that completed before our start time.
+		if out.LastUpdateTime().Before(m.params.AutoFeeStartDate) {
```

Comment:
It could still be a pending swap right?

Reply:
yeah indeed! moved the timestamp check to only apply to completed swaps, since anything that's pending is in our budget.